modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -985,13 +985,13 @@ void
 NSFVAction::addINSInitialConditions()
 {
   // do not set initial conditions if we load from file
-  if (getParam<bool>("initialize_variables_from_mesh_file"))
+  if (_app.isRestarting() || getParam<bool>("initialize_variables_from_mesh_file"))
     return;
 
   InputParameters params = _factory.getValidParams("FunctionIC");
   auto vvalue = getParam<std::vector<FunctionName>>("initial_velocity");
 
-  if (_create_velocity)
+  if (_create_velocity && (!_app.isRestarting() || _pars.isParamSetByUser("initial_velocity")))
```

Comment:
we ve already returned if app.isRestarting

I dont think this is needed

Reply:
also if we create out own velocity, we need an IC, whether the user has specified one or not, because otherwise we get a divergence with any advected scalar

Reply:
> we ve already returned if app.isRestarting

Oops that is a mistake. I need to remove the early return. Did you look at this condition closely? It says add an IC if you're not restarting or if the user has specifically requested an initial condition. You seem concerned about the former case and we will indeed add an IC, regardless of user specification, if we are not restarting. If we *are* restarting, then just like we do for non-actions, this line allows the user to override the solution vectors in the restart with an initial condition.

Reply:
ok nevermind this second comment. I guess my brain is fried from Transfers all day

Reply:
or maybe I was conceptualizing a different fix that you did. anyhow